### DISCUSSION TITLE
Discord Provider

### DISCUSSION THREAD

**Discussion by barrybtw** [nextauthjs/next-auth#4899] [Help]
### Provider type

Discord

### Environment

Environment not relevant.

### Reproduction URL

Not relevant

### Describe the issue

Some provider's like Github take in the types (string | undefined) and some just take in string, this leads to inconsistent behavior between providers. An example is the Discord provider where you have to do like this:
```
process.env.DISCORD_CLIENT as string;
process.env.DISCORD_SECRET as string;
```
And with some provider's you don't need the `as string;`.

### How to reproduce

Using a provider such as the Discord one will require you to do `as string;` and other providers not requiring it.

### Expected behavior

All provider's taking in (string | undefined)

**Comment by balazsorban44** [comment]
Not entirely sure I understand the request here, but you can type `process.env` as described here: https://stackoverflow.com/a/53981706/5364135

In any case, this is not related to NextAuth.js
